Tools/gdb/libpython.py

```diff
@@ -1502,8 +1502,10 @@ def is_python_frame(self):
         return False
 
     def is_evalframeex(self):
-        '''Is this a PyEval_EvalFrameEx frame?'''
-        if self._gdbframe.name() == 'PyEval_EvalFrameEx':
+        '''Is this a PyEval_EvalFrameEx or _PyEval_EvalFrameDefault (PEP 0523)
+        frame?'''
+        if self._gdbframe.name() in ('PyEval_EvalFrameEx',
```

Comment:
We want to keep the old name because the same python-gdb.py can be used to debug also older Python core dumps. We even keep the syntax compatible with 2.6 for it to be linkable with older gdbs. There's a comment on the top:
```
# NOTE: some gdbs are linked with Python 3, so this file should be dual-syntax
# compatible (2.6+ and 3.0+).  See #19308.
```